pytheus/registry.py

```diff
@@ -1,19 +1,25 @@
 from logging import getLogger
 from threading import Lock
-from typing import Iterable, Protocol
-
+from typing import Iterable, Optional, Protocol
 
 logger = getLogger(__name__)
 
 
 class Collector(Protocol):
     name: str
+    description: str
 
     def collect(self) -> Iterable:
         ...
 
+    @property
+    def type(self) -> str:
+        ...
+
 
 class Registry(Protocol):
+    prefix: Optional[str]
```

Comment:
I would prefer this to be the modern approach for consistency in the library `prefix: str | None`